lib/studies/reply/adaptive_nav.dart

```diff
@@ -131,6 +135,22 @@ class _DesktopNavState extends State<_DesktopNav>
           });
         }
       });
+    _destinationsCount = 0;
+
+    _indexWithDestinations = {
+      for (var destination in widget.destinations.keys) _nextInt: destination
+    };
+
+    _currentInbox = InboxPage(
+      key: UniqueKey(),
+      destination: _indexWithDestinations[widget.selectedIndex],
+    );
+  }
+
+  int get _nextInt {
```

Comment:
This feels clunky. I think `Map.forEach`, which gives you the iteration index, would be better

Reply:
It gives me the key value pair, not the index. I do agree this feels clunky though. I'll explore this some more.

Reply:
Yeah agree that maybe you can find a cleaner solution here

Reply:
Updated the implementation, got rid of counters and _nextInt function.
```dart
    final destinationKeysList = widget.destinations.keys.toList();

    _indexWithDestinations = {
      for (var i = 0; i < widget.destinations.keys.length; i++)
        i: destinationKeysList[i]
    };

    destinationKeysList.clear();
```

Reply:
Looks better! Let's see if we can improve it some more :)